lib/katello/tasks/repository.rake

```diff
@@ -65,6 +65,12 @@ namespace :katello do
         handle_missing_repo(repo)
       end
     end
+
+    ::Katello::RootRepository.find_each do |root_repo|
+      if root_repo.library_instance.nil? && root_repo.repositories.empty?
+        handle_missing_root_repo(root_repo)
+      end
+    end
```

Comment:
I think the nil check on library_instance is unnecessary if also checking that the repos are empty, in which case you could find such root repos like this: `Katello::RootRepository.where.not(id: Katello::Repository.pluck(:root_id).uniq)`

You may even want to create an `orphaned` scope on RootRepository if that works - up to you.